tests/test_cli.py

```diff
@@ -17,29 +17,111 @@
 ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
 
 
-AIRDROP_DATA = {A_ADDRESS: 10, B_ADDRESS: 20, D_ADDRESS: 30}
-
-
 def is_encoded_hash32(value: str) -> bool:
     return is_hex(value) and len(value) == 2 + 2 * 32
 
 
 @pytest.fixture()
-def airdrop_list_file(tmp_path):
+def airdrop_list_file(tmp_path, tree_data):
+    folder = tmp_path / "subfolder"
+    folder.mkdir()
+    file_path = folder / "airdrop_list.csv"
+    file_path.write_text(
+        "\n".join(
+            (
+                ",".join((to_checksum_address(address), str(value)))
+                for address, value in tree_data
+            )
+        )
+    )
+    return file_path
+
+
+@pytest.fixture()
+def wrong_airdrop_list_file(tmp_path, tree_data):
     folder = tmp_path / "subfolder"
     folder.mkdir()
     file_path = folder / "airdrop_list.csv"
     file_path.write_text(
         "\n".join(
             (
                 ",".join((to_checksum_address(address), str(value)))
-                for address, value in AIRDROP_DATA.items()
+                for address, value in tree_data[:-1]
             )
         )
     )
     return file_path
 
 
+@pytest.fixture()
+def airdrop_data(tree_data):
+    return {address: value for address, value in tree_data}
+
+
+@pytest.fixture()
+def token_contract(premint_token_value):
+    web3 = connect_to_json_rpc("test")
+    compiled_contracts = load_contracts_json("merkle_drop")
+
+    token_contract: Contract = deploy_compiled_contract(
+        abi=compiled_contracts["DroppedToken"]["abi"],
+        bytecode=compiled_contracts["DroppedToken"]["bytecode"],
+        constructor_args=(
+            "Test Token",
+            "TT",
+            18,
+            web3.eth.accounts[0],
+            premint_token_value,
+        ),
+        web3=web3,
+    )
+
+    return token_contract
+
+
+@pytest.fixture()
+def unfunded_merkle_drop_contract(
+    premint_token_value, root_hash_for_tree_data, token_contract
+):
+    web3 = connect_to_json_rpc("test")
+    compiled_contracts = load_contracts_json("merkle_drop")
+
+    merkle_drop_contract: Contract = deploy_compiled_contract(
+        abi=compiled_contracts["MerkleDrop"]["abi"],
+        bytecode=compiled_contracts["MerkleDrop"]["bytecode"],
+        constructor_args=(
+            token_contract.address,
+            premint_token_value,
+            root_hash_for_tree_data,
+            pendulum.now().int_timestamp + 10,
+            60 * 60 * 24 * 4,
+        ),
+        web3=web3,
+    )
+
+    # token_contract.functions.storeAddressOfMerkleDrop(
+    #    merkle_drop_contract.address
+    # ).transact()
```

Comment:
This comment is meant to be there? :thinking:

Reply:
I am discussing this with Come. it was something that I did not understand. This was there in Saschas code, but did not work for me.